listenbrainz/webserver/views/stats_api.py

```diff
@@ -861,22 +890,36 @@ def _get_mbids_from_msids(artist_msids: list) -> list:
             result.raise_for_status()
             data = result.json()
             for entry in data:
-                artist_mbids += entry['[artist_credit_mbid]']
+                for artist_mbid in entry['[artist_credit_mbid]']:
+                    artist_mbids.append({
+                        'artist_mbid': artist_mbid,
+                        'artist_msid': entry['artist_msid']
+                    })
         except requests.RequestException as err:
             current_app.logger.error("Error while getting artist_mbids, {}".format(err), exc_info=True)
             error_msg = ("An error occurred while calculating artist_map data, "
                          "try setting 'force_recalculate' to 'false' to get a cached copy if available."
                          "Payload: {}. Response: {}".format(request_data, result.text))
             raise APIInternalServerError(error_msg)
 
+    # Sort the result according to artist_msid and perform binary search to match with appropriate listen_count
+    artist_mbids.sort(key=lambda artist: artist['artist_msid'])
```

Comment:
instead of sorting and then applying binary search maybe it's a good idea to construct a dict (or two) which is {artist_msid/mbid: listen_count}, then you can do the lookup easily without the extra logic and bisect stuff.

it'd be much easier to read.

Reply:
Yep this makes much more sense, updated.